src/components/_global/BalTable/BalTable.vue

```diff
@@ -106,31 +106,61 @@
           >
             <td
               v-for="(column, columnIndex) in filteredColumns"
+              @click="navigate"
+              :href="href"
               :key="column.id"
               :class="[
                 column.align === 'right' ? 'text-left' : 'text-right',
                 getHorizontalStickyClass(columnIndex),
                 isColumnStuck ? 'isSticky' : ''
               ]"
             >
-              <slot
-                v-if="column.Cell"
-                v-bind="dataItem"
-                :name="column.Cell"
-              ></slot>
-              <div
-                v-else
-                :class="[
-                  'px-6 py-4',
-                  column.align === 'right' ? 'text-right' : 'text-left'
-                ]"
+              <router-link
+                v-if="link"
+                :to="{
+                  to: link.to,
+                  params: link.getParams(dataItem)
+                }"
               >
-                {{
-                  typeof column.accessor === 'string'
-                    ? dataItem[column.accessor]
-                    : column.accessor(dataItem)
-                }}
-              </div>
+                <slot
+                  v-if="column.Cell"
+                  v-bind="dataItem"
+                  :name="column.Cell"
+                ></slot>
+                <div
+                  v-else
+                  :class="[
+                    'px-6 py-4',
+                    column.align === 'right' ? 'text-right' : 'text-left'
+                  ]"
+                >
+                  {{
+                    typeof column.accessor === 'string'
+                      ? dataItem[column.accessor]
+                      : column.accessor(dataItem)
+                  }}
+                </div>
+              </router-link>
+              <template v-else>
+                <slot
+                  v-if="column.Cell"
+                  v-bind="dataItem"
+                  :name="column.Cell"
+                ></slot>
+                <div
+                  v-else
+                  :class="[
+                    'px-6 py-4',
+                    column.align === 'right' ? 'text-right' : 'text-left'
+                  ]"
+                >
+                  {{
+                    typeof column.accessor === 'string'
+                      ? dataItem[column.accessor]
+                      : column.accessor(dataItem)
+                  }}
+                </div>
+              </template>
```

Comment:
I couldn't find an elegant way to have link and non-link behaviour without duplication - 
I tried using the v-slot behaviour for custom router-links but that didn't work.
This is because for non-linkable rows we don't want users to see that you can 'right click -> open url options' in the browser but for link ones too.

Do you guys have any better ideas?

Reply:
Can we not maybe use `<component :is="tag">...` in a similar way to how we use it for `BalBtn`?

Reply:
Ill try that

Reply:
The recommended way to change the HTML element for router-links is to use the `tag` prop but that does not seem to work.

Reply:
Does this help: https://next.router.vuejs.org/guide/advanced/extending-router-link.html#extending-routerlink
Note make sure you're viewing vue-router docs @4x. A lot of google links are still pointing to @3x.

Reply:
And this: https://next.router.vuejs.org/api/#router-link-props

Reply:
Had the same problems with the new API as well. The `custom` prop you saw earlier was from the `router-link` v-slot preperation.